spec/codebreaker/game_spec.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+RSpec.describe Codebreaker::Game do
+  let(:secret_code) { %w[1 2 3 1] }
+  let(:data_i18n) { { length_good: 4, correct_range_first: 1, correct_range_last: 6 } }
+  let(:game) { described_class.new(user: Codebreaker::User.new('Anton'), type_of_difficulty: :easy) }
+  let(:my_guess_good) { game.my_guess('1231') }
+  let(:my_guess_bad) { game.my_guess('1298') }
+
+  describe '#my_guess' do
+    it 'give hash' do
+      expect(my_guess_good).to be_a_kind_of(Hash)
+    end
+
+    it 'has validation input_bad_range' do
+      expect { my_guess_bad }.to raise_error(StandardError,
+                                             I18n.t(:guess,
+                                                    length_good: data_i18n[:length_good],
+                                                    correct_range_first: data_i18n[:correct_range_first],
+                                                    correct_range_last: data_i18n[:correct_range_last]))
+    end
+
+    it 'reduce attempts' do
+      expect { my_guess_good }.to change(game, :attempts).from(15).to(14)
```

Comment:
15, 14, old_attemts_count,

let(:old_attempts_count) { 15 }

expect { my_guess_good }.to change(game, :attempts).from(old_attempts_count).to(old_attempts_count.pred)

and use context

Reply:
fixed